bindings/c/test/unit/unit_tests.cpp

```diff
@@ -0,0 +1,1576 @@
+                          (const uint8_t *)&size_limit, sizeof(size_limit)));
+  tr.set((const uint8_t *)"foo", 3, (const uint8_t *)"bar", 3);
+  fdb::EmptyFuture f1 = tr.commit();
+
+  CHECK(wait_future(f1) == 2006); // invalid_option_value
+}
+
+TEST_CASE("fdb_transaction_set_option size_limit") {
+  fdb::Transaction tr(db);
+
+  int64_t size_limit = 32;
+  fdb_check(tr.set_option(FDB_TR_OPTION_SIZE_LIMIT,
+                          (const uint8_t *)&size_limit, sizeof(size_limit)));
+  tr.set((const uint8_t *)"foo", 3,
+         (const uint8_t *)"foundation database is amazing", 30);
+  fdb::EmptyFuture f1 = tr.commit();
+
+  CHECK(wait_future(f1) == 2101); // transaction_too_large
+}
+
+// Setting the transaction size limit as a database option causes issues when
+// outside the bounds of acceptable values. TODO: Needs investigating...
+// TEST_CASE("FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT too small") {
+//   // Size limit must be at least 32 to be valid, so test a smaller size.
+//   int64_t size_limit = 31;
+//   fdb_check(fdb_database_set_option(db, FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT,
+//                                     (const uint8_t *)&size_limit,
+//                                     sizeof(size_limit)));
+//
+//   fdb::Transaction tr(db);
+//   tr.set((const uint8_t *)"foo", 3, (const uint8_t *)"bar", 3);
+//   fdb::EmptyFuture f1 = tr.commit();
+//
+//   CHECK(wait_future(f1) == 2006); // invalid_option_value
+//
+//   // Set size limit back to default.
+//   size_limit = 10000000;
+//   fdb_check(fdb_database_set_option(db, FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT,
+//                                     (const uint8_t *)&size_limit,
+//                                     sizeof(size_limit)));
+// }
+
+// TEST_CASE("FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT too large") {
+//   // Size limit must be less than or equal to 10,000,000.
+//   int64_t size_limit = 10000001;
+//   fdb_check(fdb_database_set_option(db, FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT,
+//                                     (const uint8_t *)&size_limit,
+//                                     sizeof(size_limit)));
+//
+//   fdb::Transaction tr(db);
+//   tr.set((const uint8_t *)"foo", 3, (const uint8_t *)"bar", 3);
+//   fdb::EmptyFuture f1 = tr.commit();
+//
+//   CHECK(wait_future(f1) == 2006); // invalid_option_value
+//
+//   // Set size limit back to default.
+//   size_limit = 10000000;
+//   fdb_check(fdb_database_set_option(db, FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT,
+//                                     (const uint8_t *)&size_limit,
+//                                     sizeof(size_limit)));
+// }
+
+TEST_CASE("FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT") {
+  int64_t size_limit = 32;
+  fdb_check(fdb_database_set_option(db, FDB_DB_OPTION_TRANSACTION_SIZE_LIMIT,
+                                    (const uint8_t *)&size_limit,
+                                    sizeof(size_limit)));
+
+  fdb::Transaction tr(db);
+  tr.set((const uint8_t *)"foo", 3,
```

Comment:
It might more ergonomic to have all these functions that take a pointer and a size to represent a string take a std::string_view instead. Up to you.

Reply:
I didn't know about this type, but it makes the API much nicer. Going to convert to it.